src/lib/operators/validate.cpp

```diff
@@ -56,15 +58,55 @@ std::shared_ptr<const Table> Validate::_on_execute(std::shared_ptr<TransactionCo
   DebugAssert(transaction_context->phase() == TransactionPhase::Active, "Transaction is not active anymore.");
 
   const auto in_table = input_table_left();
-
-  auto output_chunks = std::vector<std::shared_ptr<Chunk>>{};
-  output_chunks.reserve(in_table->chunk_count());
-
+  const auto chunk_count = in_table->chunk_count();
   const auto our_tid = transaction_context->transaction_id();
   const auto snapshot_commit_id = transaction_context->snapshot_commit_id();
 
-  const auto chunk_count = in_table->chunk_count();
-  for (ChunkID chunk_id{0}; chunk_id < chunk_count; ++chunk_id) {
+  std::vector<std::shared_ptr<JobTask>> jobs;
+  std::vector<std::shared_ptr<Chunk>> output_chunks;
+  output_chunks.reserve(chunk_count);
+  std::mutex output_mutex;
+
+  auto job_start_chunk_id = ChunkID{0};
+  auto job_row_count = uint32_t{0};
+  for (auto chunk_id = job_start_chunk_id; chunk_id < chunk_count; ++chunk_id) {
+    const auto chunk = in_table->get_chunk(chunk_id);
+    job_row_count += chunk->size();
+
+    // Small chunks are bundled together to avoid unnecessary scheduling overhead.
+    if (job_row_count >= Chunk::DEFAULT_SIZE || chunk_id == (chunk_count - 1)) {
+      // Single tasks are executed directly instead of scheduling a single job.
+      bool execute_directly = job_start_chunk_id == 0 && chunk_id == (chunk_count - 1);
+
+      if (execute_directly) {
+        _validate_chunks(in_table, job_start_chunk_id, chunk_id, our_tid, snapshot_commit_id, output_chunks,
```

Comment:
`job_start_chunk_id, chunk_id` - what is the second one?

Reply:
It is the `job_end_chunk_id`. But I did not want to use the name for the for-loop-variable.

Reply:
`current_chunk_id`?

Reply:
I'll change this with the `while`-loop.